engine/src/main/java/io/zeebe/engine/processor/workflow/instance/CreateWorkflowInstanceWithResultProcessor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.instance;
+
+import io.zeebe.engine.processor.KeyGenerator;
+import io.zeebe.engine.processor.TypedRecord;
+import io.zeebe.engine.processor.TypedStreamWriter;
+import io.zeebe.engine.state.deployment.WorkflowState;
+import io.zeebe.engine.state.instance.AwaitWorkflowInstanceResultMetadata;
+import io.zeebe.engine.state.instance.ElementInstanceState;
+import io.zeebe.engine.state.instance.VariablesState;
+import io.zeebe.protocol.impl.record.value.workflowinstance.WorkflowInstanceCreationRecord;
+import io.zeebe.protocol.record.RejectionType;
+import io.zeebe.protocol.record.intent.Intent;
+
+public class CreateWorkflowInstanceWithResultProcessor extends CreateWorkflowInstanceProcessor {
```

Comment:
Could you try to use composition instead of inheritance please otherwise we have here to tight coupling between these processors. But maybe we ask @saig0 what he thinks